### PR TITLE
Comparison report generation

### PR DESCRIPTION
**Pull request by karan-batavia**
Code to trigger a github action to compare the performance metrics between the stable version and the latest dev version of privado core.

### CODE REVIEW

File: .github/workflows/comparison-results.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          path: ./repos/privado-analytics
+
+      # cannot simulate restarting the terminal behavior in the pipeline, so navigated to the installation path to run the tool
+      - name: Scan karan-batavia/privado-analytics (stable) using privado scan
+        run: ~/.privado/bin/privado scan --skip-upload ${GITHUB_WORKSPACE}/repos/privado-analytics
+
+      - name: Store privado-analytics.json (stable)
+        run: mkdir ${GITHUB_WORKSPACE}/stable_results && cat ${GITHUB_WORKSPACE}/repos/privado-analytics/.privado/privado.json > ${GITHUB_WORKSPACE}/stable_results/privado-analytics.json
+
+      - name: Scan karan-batavia/privado-analytics (dev) using privado scan
```
Comment (ojaswa1942) [inline .github/workflows/comparison-results.yml:56]: Same here. Replace `karan-batavia`

File: .github/workflows/comparison-results.yml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      - name: Compare the two results
+        run: cd ${GITHUB_WORKSPACE}/tools && python ./compare.py ${GITHUB_WORKSPACE}/stable_results/privado-analytics.json ${GITHUB_WORKSPACE}/dev_results/privado-analytics.json
+
+      # For Privado-Demo/privado-accounts-api
+      - name: Cloning Privado-Demo/privado-accounts-api
+        uses: actions/checkout@v2
+        with:
+          owner: "Privado-Demo"
+          repository: "Privado-Demo/privado-accounts-api"
+          path: ./repos/privado-accounts-api
```
Comment (ojaswa1942) [inline .github/workflows/comparison-results.yml:71]: I suggest we bring these repos here soon, prefixed with `demo-` pr `sample-`, like `demo-privado-accounts-api`.

Comment (ojaswa1942) [reply .github/workflows/comparison-results.yml:71]: We can do this later too

File: .github/workflows/comparison-results.yml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        run: cd ${GITHUB_WORKSPACE}/tools && ls -lha
+
+      - name: Post results to slack
+        uses: MeilCli/slack-upload-file@v3
+        with:
+          slack_token: ${{ secrets.SLACK_TOKEN }}
+          channel_id: ${{ secrets.SLACK_CHANNEL_ID }}
+          file_path: '/home/runner/work/privado-core/privado-core/tools/comparison_report.csv'
+          initial_comment: 'Comparison Results generated.'
+          file_type: 'csv'
```
Comment (ojaswa1942) [inline .github/workflows/comparison-results.yml:122]: The check seems to be failing due to missing secrets. Let's add those first.

File: .github/workflows/comparison-results.yml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      # - name: Scan Alovoa/alovoa using privado scan
+      #   run: cd ~/.privado/bin && ./privado scan --skip-upload ${GITHUB_WORKSPACE}/repos/alovoa
+
+      # - name: Store JSON
+      #   run: cd ${GITHUB_WORKSPACE} && cat ${GITHUB_WORKSPACE}/repos/alovoa/.privado/privado.json > ${GITHUB_WORKSPACE}/stable_results/alovoa.json
+
+      # Info -> access all cloned repos at ${GITHUB_WORKSPACE}/repos
+      # Info -> access all generated JSON (stable privado) files at ${GITHUB_WORKSPACE}/stable_results
+      # Info -> access all generated JSON  (dev privado ) files at ${GITHUB_WORKSPACE}/dev_results
+      # Info -> access all the csv reports at ${GITHUB_WORKSPACE}/tools
```
Comment (ojaswa1942) [inline .github/workflows/comparison-results.yml:145]: Let's remove this if we are not using this.

**Review by ojaswa1942 — CHANGES_REQUESTED** [review]

File: .github/workflows/comparison-results.yml (hunk 1 above)
Comment (karan-batavia) [reply .github/workflows/comparison-results.yml:56]: Sure, will ask Prashant or Pandurang to add the organization name to Github secrets.

Comment (ojaswa1942) [reply .github/workflows/comparison-results.yml:56]: @karan-batavia We do not need to add that in secret. Please refer [default envs](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables) for GH actions. I'm guessing you will be using `GITHUB_REPOSITORY_OWNER`.

File: .github/workflows/comparison-results.yml (hunk 2 above)
Comment (karan-batavia) [reply .github/workflows/comparison-results.yml]: Will do, till then can't add env variable for the organization name as well

File: .github/workflows/comparison-results.yml (hunk 1 above)
Comment (karan-batavia) [reply .github/workflows/comparison-results.yml]: ```GITHUB_REPOSITORY_OWNER``` gives the current repository owner name, these repos are on different repositories

**Review by ojaswa1942 — DISMISSED** [review]
Checks failing.

**Review by prashant-privado — APPROVED** [review]

**Pull request merged**